feel-engine/src/main/scala/org/camunda/feel/interpreter/FeelInterpreter.scala

```diff
@@ -420,6 +420,11 @@ class FeelInterpreter {
       case ValDayTimeDuration(y) => ValDateTime( x.minus(y) )
       case _ => error(y, s"expect Time, or Year-Month-/Day-Time-Duration but found '$x'")
     }
+    case ValDate(x) => y match {
+      case ValDate(y) => ValYearMonthDuration( Period.between(x, y) )
+      case ValYearMonthDuration(y) => ValDate( x.minus(y) )
```

Comment:
It should also be possible to subtract a `ValDayTimeDuration`.

Reply:
Yes, agreed. I intended to do that, but I was stuck because of not knowing how to do the conversion - will add it in once I have got to grips with that.

Reply:
Hi @saig0, Should `ValDate - ValDayTimeDuration` return `ValDateTime` or `ValLocalDateTime`? What is the pattern for this?

Reply:
The result should be ValDate :smile:  For example: 
`date("2012-12-25") - duration("P1D")` = `date("2012-12-24")` 
`date("2012-12-25") - duration("P1M")` = `date("2012-11-25")`

Reply:
I'm a little confused by this reply - if removing a ValDayTimeDuration, the the user may be removing a number of hours, not a complete day, thus the result would presumably need to be a date time, not just a date?

Reply:
This is correct so far. But as a final step of the calculation, you have to transform the result back to a date - that means ignore the time part.

`date("2012-12-25") - duration("P1D")` = `date("2012-12-24")`
`date("2012-12-25") - duration("PT1H")` = `date("2012-12-24")`
`date("2012-12-25") - duration("PT1M")` = `date("2012-12-24")`
`date("2012-12-25") - duration("P1DT1H")` = `date("2012-12-23")`

Note that this feature will be part of DMN 1.2. I checked that the result should be a date.